docs/source/contributing/docs/typings.rst

```diff
@@ -91,6 +89,32 @@ Typing guidelines
 
 * Use ``typing.Iterable`` whenever the function works with *any* iterable, not a specific type.
 
+* Prefer ``numpy.typing.NDArray`` over ``numpy.ndarray``
+
+.. code:: py
+
+   import numpy as np
+
+   if TYPE_CHECKING:
+       import numpy.typing as npt
+
+
+   def foo() -> npt.NDArray[float]:
+       return np.array([1, 0, 1])
+
+* If a method returns ``self``, use ``typing_extensions.Self``.
```

Comment:
Probably worth mentioning the aliases we define? Or are they mentioned already somewhere?

Reply:
We only talk about how we have a specialized `typing` module for your hints. In regards to those, I have another PR for adding documentation about them #3669